src/main/renderers.ts

```diff
@@ -0,0 +1,58 @@
+import { app } from 'electron';
+
+import { normalizeUrl } from '../common';
+import { onWebContentsCreated } from './electron-normalize';
+
+interface Renderer {
+  id: number;
+  url?: string;
+  title?: string;
+}
+
+let RENDERERS: Map<number, Renderer> | undefined;
+
+/**
+ * Enable tracking of renderer properties via events
+ *
+ * This allows us to get the last known state of a renderer even if it's been destroyed
+ **/
+export function trackRendererProperties(): void {
+  if (RENDERERS) {
```

Comment:
this is clever to use `RENDERERS` as a flag to make sure we only add the event emitters once, but I wonder how much it reduces readability.

Wonder if an explicit boolean flag + defining `const RENDERERS = new Map<number, Renderer>();` with would help with that. I'm torn, so feel free to ignore this.

I guess this way though the map creation is lazy though, which is another added plus.

Reply:
Yes, so `trackRendererProperties` is called from two different integrations that require this data and wont be called at all if you don't use those integrations..